cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/ApiCredentials.java

```diff
@@ -0,0 +1,307 @@
+     */
+    public static Map<String, UserEntry> generateUserManagedApiCredentials(Secret secret, String secretKey) {
+        Map<String, UserEntry> entries = new HashMap<>();
+        if (secret != null) {
+            if (secretKey != null && secret.getData().containsKey(secretKey)) {
+                String credentialsAsString = Util.decodeFromBase64(secret.getData().get(secretKey));
+                entries.putAll(parseEntriesFromString(credentialsAsString));
+            }
+        }
+        for (UserEntry entry : entries.values()) {
+            if (FORBIDDEN_USERNAMES.contains(entry.username())) {
+                throw new InvalidConfigurationException("The following usernames for Cruise Control API are forbidden: " + FORBIDDEN_USERNAMES
+                        + " User provided Cruise Control API credentials contain illegal username: " + entry.username());
+            } else if (entry.role() == Role.ADMIN) {
+                throw new InvalidConfigurationException("The following roles for Cruise Control API are forbidden: " + Role.ADMIN
+                        + " User provided Cruise Control API credentials contain contains illegal role: " +  entry.role());
+            }
+        }
+        return entries;
+    }
+
+    /**
+     * Parses auth data from existing Cluster Operator managed API user secret into map of API user entries.
+     *
+     * @param passwordGenerator The password generator for API users
+     * @param secret API user secret
+     *
+     * @return Map of API user entries containing Strimzi-managed API user credentials
+     */
+    public static Map<String, UserEntry> generateCoManagedApiCredentials(PasswordGenerator passwordGenerator, Secret secret) {
+        Map<String, UserEntry> entries = new HashMap<>();
+
+        if (secret != null) {
+            if (secret.getData().containsKey(AUTH_FILE_KEY)) {
+                String credentialsAsString = Util.decodeFromBase64(secret.getData().get(AUTH_FILE_KEY));
+                entries.putAll(parseEntriesFromString(credentialsAsString));
+            }
+        }
+
+        if (!entries.containsKey(REBALANCE_OPERATOR_USERNAME)) {
+            entries.put(REBALANCE_OPERATOR_USERNAME, new UserEntry(REBALANCE_OPERATOR_USERNAME, passwordGenerator.generate(), ApiCredentials.Role.ADMIN));
+        }
+
+        if (!entries.containsKey(HEALTHCHECK_USERNAME)) {
+            entries.put(HEALTHCHECK_USERNAME, new UserEntry(HEALTHCHECK_USERNAME, passwordGenerator.generate(), ApiCredentials.Role.USER));
+        }
+
+        return entries;
+    }
+
+    /**
+     * Creates map with API usernames, passwords, and credentials file for Strimzi-managed API users secret.
+     *
+     * @param entries List of API user entries containing API user credentials
+     *
+     * @return Map containing Cruise Control API auth credentials
+     */
+    public static Map<String, String> generateMapWithApiCredentials(Map<String, UserEntry> entries) {
```

Comment:
I guess this can be private?

Reply:
Using this method in the `MockCruiseControl` class, so I prefixed the method with `/* test */`